### DISCUSSION TITLE
Saving Results from a given time step

### DISCUSSION THREAD

**Discussion by adasi11** [rsachetto/MonoAlg3D_C#50] [Q&A]
Hey, 

I was wondering whether the following utility could be easy (or reasnobale) to implement: 

Provided that a simulation reaches a given timestep, save the Vm in a similar fashion as the ECG (i.e., a text file with each row being a time step and each column an element in the mesh). So, basically, the same as the ECG function, but with the transmembrane voltage and from a given time step. 
So, the user could specify when to save results (i.e., after t = 1000 ms) and wih a print_rate that could be different from that of the [save_results]. 

Thanks,
Albert

**Comment by rsachetto** [comment]
Hi Albert, that would be easy to implement. I think I can do this later today.

My best,
Rafael Sachetto

**Reply by adasi11** [reply]
That would be fantastic, thanks!

**Reply by rsachetto** [reply]
Albert, for now, it is only possible to use the same print_rate, as the save_mesh_function is only called each dt that corresponds to the print rate. To use the new functionality just add the option **extra_function_1=save_vm_matrix** in the [save_result] section. A file called Vm_matrix.txt will be saved in the output_dir directory. You can use the start_saving_at option to define at what time this matrix will be saved. You can also use this function as a standalone save_result function.

In the future, I will add all save_functions to a static library, so it will be easy to reuse them in the custom_functions file, without duplicating code.

Let me know if this was what you needed.

My best,
Rafael
